slices/libpam-runtime.yaml

```diff
@@ -18,41 +18,123 @@ slices:
+        if section == "session-noninteractive":
+          res += "session" + upd + "\n"
+        else:
+          res += section + upd + "\n"
+        return res
+
+      def build_block(data, section, block, existing):
+        si = 1
+        res = existing
+        if section in data:
+          types = data[section]
+          if block in types:
+            # get the keys (which are the priorities) and
+            # then reverse sort them to get the highest priority
+            # first
+            itemsByPriority = types[block]
+            keys = itemsByPriority.keys()
+            keys = sorted(keys, reverse=True)
+            for p in keys:
+              for d in itemsByPriority[p]:
+                res = reconfigure_data_line(section, d, si) + "\n"
+                si += 1
+
+        # no primary block, so output a stock pam_permit line
+        # to keep the stack intact
+        if res == "" and block == "Primary":
+          return reconfigure_data_line(section, "\t[default=1]\t\t\tpam_permit.so\n", 1);
+        return res
+
+      fnames = ["account", "auth", "password", "session", "session-noninteractive"]
+      idnames = ["$account", "$auth", "$password", "$session", "$session_nonint"]
+
+      for i in range(len(fnames)):
+        fn = fnames[i]
+        template = content.read("/usr/share/pam/common-" + fn)
+        pb = build_block(confdata, fn, "Primary", "")
+        ab = build_block(confdata, fn, "Additional", "")
+
+        # session also includes settings from the session-noninteractive,
+        # but not the other way around
+        if fn == "session":
+          pb = build_block(confdata, "session-noninteractive", "Primary", pb)
+          ab = build_block(confdata, "session-noninteractive", "Additional", ab)
+
+        template = template.replace(idnames[i] + "_primary", pb)
+        template = template.replace(idnames[i] + "_additional", ab)
+        content.write("/etc/pam.d/common-" + fn, template)
 
   # folders expected by libpam to exist
   var:
     contents:
       /var/lib/pam/:
 
-  # Used by pam-auth-update to generate the pam.d/ config
-  # files inside etc. However pam-auth-update relies on
-  # debconf to actually work and discover these packages.
-  pam-config:
-    contents:
-      /usr/share/pam-configs/unix:
-
-  # default templates for the /etc/pam.d files that are used by
-  # pam-auth-update to generate the /etc/pam.d versions based on
-  # additional plugs in /usr/share/pam-configs/*.
-  pam-defaults:
-    contents:
-      /usr/share/pam/common-account:
-      /usr/share/pam/common-account.md5sums:
-      /usr/share/pam/common-auth:
-      /usr/share/pam/common-auth.md5sums:
-      /usr/share/pam/common-password:
-      /usr/share/pam/common-password.md5sums:
-      /usr/share/pam/common-session:
-      /usr/share/pam/common-session-noninteractive:
-      /usr/share/pam/common-session-noninteractive.md5sums:
-      /usr/share/pam/common-session.md5sums:
-
```

Comment:
we cannot delete slices, otherwise it will be a regression for whoever is already using them. I suggest merging this with the mutation work you did above

Reply:
But for this specific case it wouldn't make sense for someone to use these? These are used as templates or input strictly by the scripts only to generate the actual files needed - it does not make sense for these to be in the end-result.

Reply:
I can reduce them to a single `essential` that depends on config if that really is better, but I don't like that it needs to clutter for slices that were essentially "useless" for regular usage

Reply:
the slice can be modified. that's ok. but you can imagine that existing users who may have pipelines running `chisel cut ..._pam-defaults` would now have the pipeline suddenly fail for what is supposed to be a stable release. That's a regression, regardless of whether this slice makes sense. So it either stays (broken if must be) or it is fixed, ensuring the ultimate chiselled filesystem has at least the same contents as what it currently delivers.

Reply:
I just comitted db301b7d44e4361d7dd589d04a51c2ec6e5d6949 with readded slices, however the -defaults one is modified to deliver working content instead of broken. Would this be fine?